CMakeLists.txt

```diff
@@ -126,7 +144,10 @@ set_property(CACHE NMC_SYSTEM_TYPE PROPERTY STRINGS Generic Cray BGQ )
 
 # Cray specific flags
 if(${NMC_SYSTEM_TYPE} MATCHES "Cray")
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -dynamic")
+    # we no longer set the -dynamic flag for the compilere here, instead dynamic
+    # linking should be enabled by the user by setting the environment variable:
+    #   CRAYPE_LINK_TYPE=dynamic
+    #set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -dynamic")
     add_definitions(-DNMC_HAVE_CRAY)
```

Comment:
While understanding that the TBB developers make this extremely awkward, I am again worried about the use of dynamic libraries in environments where the build file system is not available on the compute nodes.